Compilation failure

## Error message

```
src/indexer.cc:261:37: error: variable length array of non-POD element type 'intrusive_ptr<IndexKey>'
        intrusive_ptr<IndexKey> keys[nrunes];
                                    ^
```

## GCC Version

```
[~/m/livegrep]─[⎇ master]─(1)-> gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.38)
Target: x86_64-apple-darwin16.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
<details>

<summary>All logs</summary>

```
[~/m/livegrep]─[⎇ master]─> bazel build //...
INFO: Found 32 targets...
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_file.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_file.cc:159:6: warning: unused function 'CompareFieldsByName' [-Wunused-function]
bool CompareFieldsByName(const FieldDescriptor *a, const FieldDescriptor *b) {
     ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/cpp/cpp_message.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/cpp/cpp_message.cc:376:8: warning: unused function 'MessageTypeProtoName' [-Wunused-function]
string MessageTypeProtoName(const FieldDescriptor* field) {
       ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/field_mask_utility.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/util/internal/field_mask_utility.cc:47:14: warning: unused function 'CreatePublicError' [-Wunused-function]
util::Status CreatePublicError(util::error::Code code,
             ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/utility.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/util/internal/utility.cc:52:19: warning: unused function 'SkipWhiteSpace' [-Wunused-function]
const StringPiece SkipWhiteSpace(StringPiece str) {
                  ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_enum_lite.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_enum_lite.cc:53:6: warning: unused function 'EnumHasCustomOptions' [-Wunused-function]
bool EnumHasCustomOptions(const EnumDescriptor* descriptor) {
     ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/js/js_generator.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/js/js_generator.cc:249:8: warning: unused function 'GetPath' [-Wunused-function]
string GetPath(const GeneratorOptions& options,
       ^
external/com_github_google_protobuf/src/google/protobuf/compiler/js/js_generator.cc:266:8: warning: unused function 'GetPath' [-Wunused-function]
string GetPath(const GeneratorOptions& options,
       ^
external/com_github_google_protobuf/src/google/protobuf/compiler/js/js_generator.cc:547:8: warning: unused function 'JSMapGetterName' [-Wunused-function]
string JSMapGetterName(const GeneratorOptions& options,
       ^
3 warnings generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_message_builder_lite.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_message_builder_lite.cc:71:8: warning: unused function 'MapValueImmutableClassdName' [-Wunused-function]
string MapValueImmutableClassdName(const Descriptor* descriptor,
       ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_message_lite.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/compiler/java/java_message_lite.cc:1032:6: warning: unused function 'CheckHasBitsForEqualsAndHashCode' [-Wunused-function]
bool CheckHasBitsForEqualsAndHashCode(const FieldDescriptor* field) {
     ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/text_format.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/text_format.cc:77:13: warning: unused function 'GetAnyFieldDescriptors' [-Wunused-function]
inline bool GetAnyFieldDescriptors(const Message& message,
            ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:52:18: warning: unused variable 'kMicrosPerMillisecond' [-Wunused-const-variable]
static const int kMicrosPerMillisecond = 1000;
                 ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:56:19: warning: unused variable 'kTimestampFormat' [-Wunused-const-variable]
static const char kTimestampFormat[] = "%E4Y-%m-%dT%H:%M:%S";
                  ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:380:6: warning: unused function 'ToUint128' [-Wunused-function]
void ToUint128(const Timestamp& value, uint128* result, bool* negative) {
     ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:405:6: warning: unused function 'ToTimestamp' [-Wunused-function]
void ToTimestamp(const uint128& value, bool negative, Timestamp* timestamp) {
     ^
4 warnings generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:90:13: warning: unused function 'IsHighSurrogate' [-Wunused-function]
inline bool IsHighSurrogate(uint16 c) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:102:13: warning: unused function 'IsLowSurrogate' [-Wunused-function]
inline bool IsLowSurrogate(uint16 c) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:122:13: warning: unused function 'IsSupplementalCodePoint' [-Wunused-function]
inline bool IsSupplementalCodePoint(uint32 cp) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:138:15: warning: unused function 'ToCodePoint' [-Wunused-function]
inline uint32 ToCodePoint(uint16 high, uint16 low) {
              ^
4 warnings generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/generated_message_reflection.cc [for host]:
external/com_github_google_protobuf/src/google/protobuf/generated_message_reflection.cc:76:13: warning: unused function 'SupportsArenas' [-Wunused-function]
inline bool SupportsArenas(const Descriptor* descriptor) {
            ^
1 warning generated.
INFO: From Compiling external/com_github_libgit2/deps/zlib/inflate.c:
external/com_github_libgit2/deps/zlib/inflate.c:1507:61: warning: shifting a negative signed value is undefined [-Wshift-negative-value]
    if (strm == Z_NULL || strm->state == Z_NULL) return -1L << 16;
                                                        ~~~ ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:90:13: warning: unused function 'IsHighSurrogate' [-Wunused-function]
inline bool IsHighSurrogate(uint16 c) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:102:13: warning: unused function 'IsLowSurrogate' [-Wunused-function]
inline bool IsLowSurrogate(uint16 c) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:122:13: warning: unused function 'IsSupplementalCodePoint' [-Wunused-function]
inline bool IsSupplementalCodePoint(uint32 cp) {
            ^
external/com_github_google_protobuf/src/google/protobuf/util/internal/json_escaping.cc:138:15: warning: unused function 'ToCodePoint' [-Wunused-function]
inline uint32 ToCodePoint(uint16 high, uint16 low) {
              ^
4 warnings generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/utility.cc:
external/com_github_google_protobuf/src/google/protobuf/util/internal/utility.cc:52:19: warning: unused function 'SkipWhiteSpace' [-Wunused-function]
const StringPiece SkipWhiteSpace(StringPiece str) {
                  ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/generated_message_reflection.cc:
external/com_github_google_protobuf/src/google/protobuf/generated_message_reflection.cc:76:13: warning: unused function 'SupportsArenas' [-Wunused-function]
inline bool SupportsArenas(const Descriptor* descriptor) {
            ^
1 warning generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:52:18: warning: unused variable 'kMicrosPerMillisecond' [-Wunused-const-variable]
static const int kMicrosPerMillisecond = 1000;
                 ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:56:19: warning: unused variable 'kTimestampFormat' [-Wunused-const-variable]
static const char kTimestampFormat[] = "%E4Y-%m-%dT%H:%M:%S";
                  ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:380:6: warning: unused function 'ToUint128' [-Wunused-function]
void ToUint128(const Timestamp& value, uint128* result, bool* negative) {
     ^
external/com_github_google_protobuf/src/google/protobuf/util/time_util.cc:405:6: warning: unused function 'ToTimestamp' [-Wunused-function]
void ToTimestamp(const uint128& value, bool negative, Timestamp* timestamp) {
     ^
4 warnings generated.
INFO: From Compiling external/com_github_google_protobuf/src/google/protobuf/util/internal/field_mask_utility.cc:
external/com_github_google_protobuf/src/google/protobuf/util/internal/field_mask_utility.cc:47:14: warning: unused function 'CreatePublicError' [-Wunused-function]
util::Status CreatePublicError(util::error::Code code,
             ^
1 warning generated.
INFO: From Linking external/com_googlesource_code_re2/libre2.so:
clang: warning: argument unused during compilation: '-pthread'
INFO: From Linking external/boost/libsmart_ptr.a:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/boost/_objs/smart_ptr/external/boost/libs/smart_ptr/src/sp_collector.pic.o has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/boost/_objs/smart_ptr/external/boost/libs/smart_ptr/src/sp_debug_hooks.pic.o has no symbols
warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning for library: bazel-out/local-fastbuild/bin/external/boost/libsmart_ptr.a the table of contents is empty (no object file members in the library define global symbols)
INFO: From Linking external/com_github_google_protobuf/libprotobuf_lite.a [for host]:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/host/bin/external/com_github_google_protobuf/_objs/protobuf_lite/external/com_github_google_protobuf/src/google/protobuf/stubs/atomicops_internals_x86_gcc.o has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/host/bin/external/com_github_google_protobuf/_objs/protobuf_lite/external/com_github_google_protobuf/src/google/protobuf/stubs/atomicops_internals_x86_msvc.o has no symbols
INFO: From Linking external/com_github_json_c/libjson.a:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/com_github_json_c/_objs/json/external/com_github_json_c/json-c/libjson.pic.o has no symbols
INFO: From Linking external/com_github_libgit2/liblibgit2.a:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/com_github_libgit2/_objs/libgit2/external/com_github_libgit2/src/stransport_stream.pic.o has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/com_github_libgit2/_objs/libgit2/external/com_github_libgit2/src/transports/auth_negotiate.pic.o has no symbols
INFO: From Linking external/boost/libfilesystem.a:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/local-fastbuild/bin/external/boost/_objs/filesystem/external/boost/libs/filesystem/src/windows_file_codecvt.pic.o has no symbols
INFO: From Linking external/com_github_google_protobuf/libprotobuf.a [for host]:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/host/bin/external/com_github_google_protobuf/_objs/protobuf/external/com_github_google_protobuf/src/google/protobuf/io/gzip_stream.o has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: bazel-out/host/bin/external/com_github_google_protobuf/_objs/protobuf/external/com_github_google_protobuf/src/google/protobuf/util/internal/error_listener.o has no symbols
ERROR: /Users/joel/me/livegrep/src/BUILD:1:1: C++ compilation of rule '//src:codesearch' failed: Process exited with status 1 [sandboxed].
src/indexer.cc:71:24: warning: expression result unused [-Wunused-value]
    edges_.insert(val).first;
    ~~~~~~~~~~~~~~~~~~ ^~~~~
src/indexer.cc:261:37: error: variable length array of non-POD element type 'intrusive_ptr<IndexKey>'
        intrusive_ptr<IndexKey> keys[nrunes];
                                    ^
1 warning and 1 error generated.
Use --strategy=CppCompile=standalone to disable sandboxing for the failing actions.
INFO: Elapsed time: 685.630s, Critical Path: 130.93s
```

</details>

--- a/src/indexer.cc
+++ b/src/indexer.cc
@@ -258,9 +258,9 @@ namespace {
     intrusive_ptr<IndexKey> CaseFoldLiteral(Rune *runes, int nrunes) {
         if (nrunes == 0)
             return Empty();
-        intrusive_ptr<IndexKey> keys[nrunes];
+        std::vector<intrusive_ptr<IndexKey> > keys;
         for (int i = 0; i < nrunes; ++i) {
-            keys[i] = CaseFoldLiteral(runes[i]);
+            keys.push_back(CaseFoldLiteral(runes[i]));
         }
         return Concat(&keys[0], nrunes);
     }

